src/olympia/blocklist/tasks.py

```diff
@@ -144,9 +151,34 @@ def import_block_from_blocklist(record):
     else:
         kinto_import.outcome = KintoImport.OUTCOME_NOMATCH
         log.debug('Kinto %s: No addon found', kinto_id)
+    if not import_created:
+        # now reconcile the blocks that were connected to the import last time
+        # but weren't changed this time - i.e. blocks we need to delete
+        delete_qs = (
+            Block.objects.filter(id__in=existing_block_ids)
+                         .exclude(id__in=(block.id for block in new_blocks)))
+        for block in delete_qs:
+            block_activity_log_delete(
+                block, delete_user=block_kw['updated_by'])
+            block.delete()
+
     kinto_import.save()
 
 
+@task
+@use_primary_db
+@transaction.atomic
+def delete_imported_block_from_blocklist(kinto_id):
+    existing_blocks = (
+        Block.objects.filter(kinto_id__in=(kinto_id, f'*{kinto_id}')))
```

Comment:
Why do we have to prepend the id with `*`?

Reply:
We don't have to (it's an `__in` with `kinto_id` too) but it's one of the ways.  Prefixing with `*` is the magic code to indicate that the Block came from a regular expression in the guid field in a v2 block so they'll be (potentially) multiple Blocks on AMO from a single record in the v2 blocklist.

Reply:
See `Block.is_imported_from_kinto_regex`

Reply:
ah thanks, I imagined something like this but I did not know the reasons behind that.